### PR TITLE
CLI command redesign for less redundancy.

### PR DESCRIPTION
**Pull request by isht3**
- `evalai challenges`
- `evalai challenges ongoing`
- `evalai challenges past`
- `evalai challenges future`
- `evalai challenges --host`
- `evalai challenges --participant`

### CODE REVIEW

File: evalai/challenges.py [hunk 1]
```diff
@@ -11,66 +11,43 @@
...
     """
     Used to list challenges.
-    Invoked by running `evalai challenges list`
+    Invoked by running `evalai challenges`
     """
-    if participant:
-        get_challenge_count(host, participant)
-    elif host:
+    if participant or host:
         get_challenge_count(host, participant)
```
Comment (varunagrawal) [inline evalai/challenges.py]: I'm not sure the naming of this function makes sense. It makes it seem like it is returning a number.

File: evalai/challenges.py [hunk 2]
```diff
@@ -11,66 +11,43 @@
...
+def challenges(ctx, participant, host):
     """
     Used to list challenges.
-    Invoked by running `evalai challenges list`
+    Invoked by running `evalai challenges`
     """
-    if participant:
-        get_challenge_count(host, participant)
-    elif host:
+    if participant or host:
```
Comment (varunagrawal) [inline evalai/challenges.py]: So here's a potential bug. The function being called when either participant or host is True first checks for host and if host is false, checks for participant.
Now if I activate both flags, I'll only ever get host challenges. You'll have to redesign the `get_challenge_count` function to account for this (and possibly rename it too).

Comment (isht3) [reply evalai/challenges.py]: @varunagrawal I've added an issue for this. I'll resolve it.

File: evalai/challenges.py (hunk 1 above)
Comment (varunagrawal) [reply evalai/challenges.py]: @isht3 any updates on this? Else everything else looks good.

Comment (isht3) [reply evalai/challenges.py]: No, not on this PR anymore. I've made an issue which I would go back to after the primary features are done.

File: evalai/challenges.py [hunk 3]
```diff
@@ -11,66 +11,43 @@
...
     elif ctx.invoked_subcommand is None:
         get_challenge_list()
 
 
-@click.command(name='ongoing')
-def list_ongoing_challenges():
+@challenges.command()
+def ongoing():
     """
     Used to list all the challenges which are active.
```
Comment (RishabhJain2018) [inline evalai/challenges.py]: screen shot 2018-06-10 at 4 13 18 pm
I think either we need to shorten the text or increase the word limit.

**Comment by isht3** [comment]
@RishabhJain2018 Just started a new issue #35. I'll rectify it.

**Comment by isht3** [comment]
@varunagrawal Can you please check this?

File: evalai/challenges.py (hunk 1 above)
Comment (RishabhJain2018) [reply evalai/challenges.py]: @isht3 I think we should fix this here only. Can you please add it quickly?

**Comment by RishabhJain2018** [comment]
The changes for this PR has been taken into #44 . Hence, closing this one.

**Pull request closed without merging**